oximeter/instruments/src/kstat/sampler.rs

```diff
@@ -1250,9 +1310,21 @@ impl oximeter::Producer for KstatSampler {
         loop {
             match rx.try_recv() {
                 Ok(sample) => samples.push(sample),
-                Err(mpsc::error::TryRecvError::Empty) => break,
-                Err(mpsc::error::TryRecvError::Disconnected) => {
-                    panic!("kstat stampler self-stat queue tx disconnected");
+                Err(broadcast::error::TryRecvError::Empty) => break,
+                Err(broadcast::error::TryRecvError::Lagged(n_missed)) => {
+                    warn!(
+                        self.log,
+                        "producer missed some samples because they're \
+                        being produced faster than `oximeter` is collecting";
+                        "n_missed" => %n_missed,
+                    );
+                }
```

Comment:
should we be recording missing samples in this case? it would be nice to let Oximeter know we've dropped some stuff...

Reply:
Ah, yeah, if it's a bigger change, let's do that separately. I do think it's worth considering this kind of data loss as a first-class construct in Oximeter, but that seems like its own chunk of work.

Reply:
>  I do think it's worth considering this kind of data loss as a first-class construct in Oximeter, but that seems like its own chunk of work.

`oximeter` does already have the ability to represent missing samples. The issue is that at this point, we don't know which timeseries the missing samples belongs to because we don't have the sample. We need a _new_ timeseries, specifically something like `kstat_sampler:self_stat_samples_dropped` or similar, to represent this.

Reply:
Tracked in https://github.com/oxidecomputer/omicron/issues/8025